### ISSUE TITLE
Attribute error on cdscan

### ISSUE THREAD

**Issue by pochedls** [CDAT/cdms#256]
**I'm using Python 3.6 with CDAT8** (on crunchy): 

`conda create -n cdat8 -c conda-forge -c cdat python=3.6 cdat mesalib joblib mysql-connector-python scandir`

**When I try to scan some directories**:

`cdscan -x test.xml /p/css03/cmip5_css01/data/cmip5/output1/CNRM-CERFACS/CNRM-CM5-2/historical/fx/ocean/fx/r0i0p0/v20130401/areacello/*.nc`

**I get a cdscan error**:

`Finding common directory ...
Common directory: /p/css03/cmip5_css01/data/cmip5/output1/CNRM-CERFACS/CNRM-CM5-2/historical/fx/ocean/fx/r0i0p0/v20130401/areacello/
Scanning files ...
/p/css03/cmip5_css01/data/cmip5/output1/CNRM-CERFACS/CNRM-CM5-2/historical/fx/ocean/fx/r0i0p0/v20130401/areacello/areacello_fx_CNRM-CM5-2_historical_r0i0p0.nc
Traceback (most recent call last):
  File "/export_backup/pochedley1/bin/anaconda2/envs/cdat8/bin/cdscan", line 1840, in <module>
    main(sys.argv)
  File "/export_backup/pochedley1/bin/anaconda2/envs/cdat8/bin/cdscan", line 1153, in main
    varnames.append(axisvar.id)
AttributeError: 'dict_keys' object has no attribute 'append'`

**Comment by pochedls** [comment]
Sorry - I realize this is likely the same as #239. @dnadeau4 - this ticket could probably be merged / closed as a duplicate of #239.

**Comment by dnadeau4** [comment]
@pochedls I am on travel next week.  I will look at this issue the week after.

**Comment by dnadeau4** [comment]
@pochedls 

In cdscan you can add list() to `f.variables.keys()`.  That is a py2 to py3 difference that is not easy to catch.

```
1134         varnames = list(f.variables.keys())
1135 
1136         # Try to force all axes to be included, but only small ones, length<100.
1137         # This section was motivated by a need to preserve the cloud axes isccp_prs,isccp_tau.
1138         # If we ever need to preserve longer axes as well, we could create one
1139         # variable per axis...
```

**Comment by pochedls** [comment]
I modified my cdscan file and this is working now. Thanks @dnadeau4

**Comment by dnadeau4** [comment]
I will let this open and create a PR and merge before closing.  Thanks!

**Comment by mzelinka** [comment]
the fix recommended by @dnadeau4 from 8/14/18 also worked like a charm for me.

**Comment by durack1** [comment]
@forsyth2 @jasonb5 this might be another issue that can be closed due to #399?

**Comment by jasonb5** [comment]
@durack1 Thank you, this is definitely another duplicate and will close.
